jetstream/argo.py

```diff
@@ -89,7 +89,20 @@ def submit_workflow(
                     failed = True
 
             time.sleep(1)
-    return not failed
+
+    # check status of pods
+    all_pods_succeeded = True
+
+    if workflow.status and workflow.status.nodes:
+        all_pods_succeeded = all(
+            [
+                node.phase == "Succeeded"
+                for _, node in workflow.status.nodes.items()
+                if node.type == "Pod"
+            ]
+        )
+
+    return not failed and all_pods_succeeded
```

Comment:
hm, `failed` is always False here, isn't it? Otherwise we'd have terminated by raising an exception. So we can probably drop the failed flag.